clients/graphql-kotlin-client-serialization/src/main/kotlin/com/expediagroup/graphql/client/serialization/GraphQLClientKotlinxSerializer.kt

```diff
@@ -43,7 +44,14 @@ class GraphQLClientKotlinxSerializer(private val jsonBuilder: JsonBuilder.() ->
         encodeDefaults = true
     }
 
-    override fun serialize(request: Any): String = json.encodeToString(AnyKSerializer, request)
+    override fun serialize(request: GraphQLClientRequest<*>): String = json.encodeToString(requestSerializer(request), request)
+
+    override fun serialize(requests: List<GraphQLClientRequest<*>>): String {
+        val serializedRequests = requests.map { request ->
```

Comment:
Kotlinx serializer is pretty bad with dynamic types - this was the simplest way I could think of how to correctly serialize list of generic types